### PR TITLE
{2023.06}[GCCcore/12.3.0] at-spi2-core 2.49.91

### LINKED ISSUE #196: Issues lurking with `gobject-introspection`
With something that was using Meson I was seeing
```
/bin/bash: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /cvmfs/pilot.eessi-hpc.org/versions/2021.12/compat/linux/x86_64/usr/lib64/libtinfo
.so.6)
```
which I'm guessing meant it was picking the host `ld` and not the EasyBuild wrapper. Looking at the configure step I saw
```
C compiler for the host machine: gcc (gcc 10.3.0 "gcc (GCC) 10.3.0")
C linker for the host machine: gcc ld.gold 2.37
```
which would imply it is using the ld.gold used by gcc (which seems right since it reports 2.37, the version in the prefix, not 2.30 the version in the host).

Anyway, it seemed like a good idea to set the linker directly ([as per the docs](https://mesonbuild.com/Reference-tables.html#compiler-and-linker-selection-variables)) with
```
preconfigopts = "export CC_LD=$(which ld) &&"
```
but actually this throws an error
```
== 2022-11-08 12:59:53,619 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/software/EasyBuild/4.6.0/lib/python3.9/site-packages/easybuild/base/exceptions.py:124 in __init__): cmd "export CC_LD=$(which ld) && meson --prefix /project/def-sponsor00/easybuild/software/at-spi2-core/2.40.2-GCCcore-10.3.0 --libdir lib  /project/60005/easybuild/build/atspi2core/2.40.2/GCCcore-10.3.0/at-spi2-core-2.40.2/" exited with exit code 1 and output:
The Meson build system
Version: 0.58.0
Source dir: /project/60005/easybuild/build/atspi2core/2.40.2/GCCcore-10.3.0/at-spi2-core-2.40.2
Build dir: /project/60005/easybuild/build/atspi2core/2.40.2/GCCcore-10.3.0/easybuild_obj
Build type: native build
Project name: at-spi2-core
Project version: 2.40.2

../at-spi2-core-2.40.2/meson.build:1:0: ERROR: Unsupported linker, only bfd, gold, and lld are supported, not /tmp/eb-nf5k_pyz/tmp1a3n724j/rpath_wrappers/ld_wrapper/ld.
```

### PR DESCRIPTION
**Pull request by bedroge**
Split off from #428, trying to fix the `at-spi2-core` issue (and should fix #196 for `at-spi2-core`, though other apps depending on `GObject-Introspection` could have similar issues). Also see https://gitlab.com/eessi/support/-/issues/29.

```
1 out of 32 required modules missing:

* at-spi2-core/2.49.91-GCCcore-12.3.0 (at-spi2-core-2.49.91-GCCcore-12.3.0.eb)
```

### CODE REVIEW

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Review by ocaisa — APPROVED** [review]

File: eb_hooks.py [hunk 1]
```diff
@@ -335,6 +335,20 @@ def pre_configure_hook_LAMMPS_aarch64(self, *args, **kwargs):
         raise EasyBuildError("LAMMPS-specific hook triggered for non-LAMMPS easyconfig?!")
 
 
+def pre_configure_hook_atspi2core_filter_ld_library_path(self, *args, **kwargs):
+    """
+    pre-configure hook for at-spi2-core:
+    - instruct GObject-Introspection's g-ir-scanner tool to not set $LD_LIBRARY_PATH
+      when EasyBuild is configured to filter it
```
Comment (boegel) [inline eb_hooks.py:342]: let's mention https://github.com/EESSI/software-layer/issues/196 in here too?

Comment (bedroge) [reply eb_hooks.py]: done

**Comment by boegel** [comment]
all staging PRs merged...

**Review by boegel — APPROVED** [review]

**Pull request merged**